test/test_euler.py

```diff
@@ -0,0 +1,770 @@
+    #            eoc_rec1 = EOCRecorder()
+
+#    for nel_1d in [4, 8, 12]:
+    for nel_1d in [8]:
+        from meshmode.mesh.generation import generate_regular_rect_mesh
+        mesh = generate_regular_rect_mesh(
+            a=(-0.5,) * dim, b=(0.5,) * dim, n=(nel_1d,) * dim
+        )
+
+        logger.info(
+            f"Number of {dim}d elements: {mesh.nelements}"
+        )
+
+        discr = EagerDGDiscretization(actx, mesh, order=order)
+
+        mass_input = discr.zeros(actx) + 1
+        energy_input = discr.zeros(actx) + 2.5
+
+        mom_input = make_obj_array(
+            [discr.zeros(actx) for i in range(discr.dim)]
+        )
+        fields = flat_obj_array(
+            mass_input, energy_input, mom_input
+        )
+
+        expected_rhs = make_obj_array(
+            [discr.zeros(actx) for i in range(discr.dim + 2)]
+        )
+
+        boundaries = {BTAG_ALL: DummyBoundary()}
+        inviscid_rhs = inviscid_operator(discr, fields, boundaries)
+        rhs_resid = inviscid_rhs - expected_rhs
+
+        resid_split = split_conserved(dim, rhs_resid)
+        rho_resid = resid_split.mass
+        rhoe_resid = resid_split.energy
+        mom_resid = resid_split.momentum
+
+        rhs_split = split_conserved(dim, inviscid_rhs)
+        rho_rhs = rhs_split.mass
+        rhoe_rhs = rhs_split.energy
+        rhov_rhs = rhs_split.momentum
+
+        message = (
+            f"rho_rhs  = {rho_rhs}\n"
+            f"rhoe_rhs = {rhoe_rhs}\n"
+            f"rhov_rhs = {rhov_rhs}"
+        )
+        logger.info(message)
+
+        assert discr.norm(rho_resid, np.inf) < tolerance
+        assert discr.norm(rhoe_resid, np.inf) < tolerance
+        for i in range(dim):
+            assert discr.norm(mom_resid[i], np.inf) < tolerance
+
+            err_max = discr.norm(rhs_resid[i], np.inf)
+            #                eoc_rec0.add_data_point(1.0 / nel_1d, err_max)
+            assert(err_max < tolerance)
+            if err_max > maxxerr:
+                maxxerr = err_max
+        # set a non-zero, but uniform velocity component
+
+        for i in range(len(mom_input)):
+            mom_input[i] = discr.zeros(actx) + (-1.0) ** i
+
+        boundaries = {BTAG_ALL: DummyBoundary()}
+        inviscid_rhs = inviscid_operator(discr, fields, boundaries)
+        rhs_resid = inviscid_rhs - expected_rhs
+
+        resid_split = split_conserved(dim, rhs_resid)
+        rho_resid = resid_split.mass
+        rhoe_resid = resid_split.energy
+        mom_resid = resid_split.momentum
+
+        assert discr.norm(rho_resid, np.inf) < tolerance
+        assert discr.norm(rhoe_resid, np.inf) < tolerance
+
+        for i in range(dim):
+            assert discr.norm(mom_resid[i], np.inf) < tolerance
+            err_max = discr.norm(rhs_resid[i], np.inf)
+            #                eoc_rec1.add_data_point(1.0 / nel_1d, err_max)
+            assert(err_max < tolerance)
+            if err_max > maxxerr:
+                maxxerr = err_max
```

Comment:
`discr.norm` takes the norm (in the case of the inf-norm, the maximum) over object array entries, so this should be simplified.

Reply:
When it fails, I want to know which components have the problem.